Assets/MixedRealityToolkit/Providers/BaseSpatialMeshObserver.cs

```diff
@@ -0,0 +1,279 @@
+﻿// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using System.Collections.Generic;
+using UnityEngine;
+using UnityEngine.EventSystems;
+
+namespace Microsoft.MixedReality.Toolkit.SpatialAwareness
+{
+    /// <summary>
+    /// Class providing a base implementation of the <see cref="IMixedRealitySpatialAwarenessMeshObserver"/> interface.
+    /// </summary>
+    public abstract class BaseSpatialMeshObserver : BaseSpatialObserver, IMixedRealitySpatialAwarenessMeshObserver
```

Comment:
It's not exactly clear the difference between BaseSpatialMeshObserver  and BaseSpatialObserver? Why are they separated?

Reply:
There's also a BaseSpatialPlaneObserver, and there could be one for point clouds or any arbitrary spatial representation

Reply:
I tried to keep BaseSpatialObserver as generic as it already was, with some additional functionality. Everything mesh-specific went into the mesh one.